zone/command.cpp

```diff
@@ -5167,116 +5167,155 @@ void command_findnpctype(Client *c, const Seperator *sep)
-					uint32 charid = c->GetTarget()->CastToClient()->CharacterID();
-					uint32 factionid = atoi(faction_filter.c_str());
 
-					if (c->GetTarget()->CastToClient()->ReloadCharacterFaction(c->GetTarget()->CastToClient(), factionid, charid))
-						c->Message(Chat::Yellow, "faction %u was cleared.", factionid);
-					else
-						c->Message(Chat::Red, "An error occurred clearing faction %u", factionid);
-				}
-				else
-				{
-					c->Message(Chat::Red, "Cannot be in Combat");
+		auto faction_message = (
+			found_count > 0 ? 
+			(
+				found_count == 1 ?
+				"A Faction was" :
+				fmt::format("{} Factions were", found_count)
+			) :
+			"No Factions were"
+		);
+		c->Message(
+			Chat::White,
+			fmt::format(
+				"{} found.",
+				faction_message
+			).c_str()
+		);
+	} else if (!strcasecmp(sep->arg[1], "reset")) {
+		if (strlen(faction_filter.c_str()) > 0) {
+			if (c->GetTarget() && c->GetTarget()->IsClient()) {
+				Client* target = c->GetTarget()->CastToClient();
+				if (
+					(
+						!c->GetFeigned() &&
+						c->GetAggroCount() == 0
+					) ||
+					(
+						!target->GetFeigned() &&
+						target->GetAggroCount() == 0
+					)
+				) {
+					uint32 character_id = target->CharacterID();
+					uint32 faction_id = std::stoul(faction_filter.c_str());
+					if (target->ReloadCharacterFaction(target, faction_id, character_id)) {
+						c->Message(
+							Chat::White,
+							fmt::format(
+								"Faction Reset | {} ({}) was reset for {}.",
+								content_db.GetFactionName(faction_id),
+								faction_id,
+								target->GetCleanName()
+							).c_str()
+						);
+					} else {
+						c->Message(
+							Chat::White,
+							fmt::format(
+								"Faction Reset Failed | {} ({}) was unable to be reset for {}.",
+								content_db.GetFactionName(faction_id),
+								faction_id,
+								target->GetCleanName()
+							).c_str()
+						);
+					}
+				} else {
+					c->Message(Chat::White, "You cannot reset factions while you or your target is in combat or feigned.");
 					return;
 				}
-			}
-			else {
-				c->Message(Chat::Red, "Player Target Required (whose not feigning death)");
+			} else {
+				c->Message(Chat::White, "You must target a PC for this command.");
 				return;
 			}
+		} else {
+			c->Message(Chat::White, "Usage: #faction reset [Faction ID] - Reset Targeted Player's Faction to Base Faction Value");
+		}
+	} else if (!strcasecmp(sep->arg[1], "view")) {
+		if (c->GetTarget() && c->GetTarget()->IsNPC()) {
+			Mob* target = c->GetTarget();
+			uint32 npc_id = target->GetNPCTypeID();
+			uint32 npc_faction_id = target->CastToNPC()->GetPrimaryFaction();
+			std::string npc_name = target->GetCleanName();
+			c->Message(
+				Chat::White,
+				fmt::format(
+					"{} ({}) has a Primary Faction  of {} ({}).",
+					npc_name,
+					npc_id,
+					content_db.GetFactionName(npc_faction_id),
+					npc_faction_id
+				).c_str()
+			);
+			c->Message(Chat::White, "Use: #setfaction [Faction ID] - Set an NPC's Primary Faction ID");
```

Comment:
Not sure of the need for #setfaction message here

Reply:
Yeah, it was there before so I just left it. I can remove it.